mod-pubsub-server/src/main/java/org/folio/services/impl/SecurityManagerImpl.java

```diff
@@ -206,71 +196,60 @@ public SecurityManagerImpl(@Autowired Vertx vertx, @Autowired Cache cache,
       });
   }
 
-  private Future<Boolean> assignPermissions(String userId, OkapiConnectionParams params) {
-    List<String> permissions = readPermissionsFromResource(PERMISSIONS_FILE_PATH);
-    if (CollectionUtils.isEmpty(permissions)) {
-      LOGGER.info("No permissions found to assign to {} user", systemUserConfig.getName());
-      return Future.succeededFuture(false);
-    }
+  private Future<Void> assignPermissions(String userId, OkapiConnectionParams params) {
     JsonObject requestBody = new JsonObject()
       .put("id", UUID.randomUUID().toString())
       .put("userId", userId)
-      .put("permissions", new JsonArray(permissions));
+      .put("permissions", new JsonArray(PERMISSIONS));
     return doRequest(params, PERMISSIONS_URL, HttpMethod.POST, requestBody.encode())
       .compose(response -> {
-        Promise<Boolean> promise = Promise.promise();
         if (response.getCode() == HttpStatus.HTTP_CREATED.toInt()) {
-          LOGGER.info("Added permissions [{}] for {} user", StringUtils.join(permissions, ","),
+          LOGGER.info("Added permissions [{}] for {} user", StringUtils.join(PERMISSIONS, ","),
             systemUserConfig.getName());
-          promise.complete(true);
+          return Future.succeededFuture();
         } else {
           String errorMessage = format("Failed to add permissions for %s user. Received status code %s",
             systemUserConfig.getName(), response.getCode());
           LOGGER.error(errorMessage);
-          promise.fail(errorMessage);
+          return Future.failedFuture(errorMessage);
         }
-        return promise.future();
       });
   }
 
-  private Future<Boolean> addPermissions(String userId, OkapiConnectionParams params) {
-    List<String> permissions = readPermissionsFromResource(PERMISSIONS_FILE_PATH);
-    if (CollectionUtils.isEmpty(permissions)) {
-      LOGGER.info("No permissions found to add for {} user", systemUserConfig.getName());
-      return Future.succeededFuture(false);
-    }
-    List<Future<?>> futures = new ArrayList<>();
+  private Future<Void> addPermissions(String userId, OkapiConnectionParams params) {
+    List<Future<Void>> futures = new ArrayList<>();
     String permUrl = PERMISSIONS_URL + "/" + userId + "/permissions?indexField=userId";
-    permissions.forEach(permission -> {
+    PERMISSIONS.forEach(permission -> {
       JsonObject requestBody = new JsonObject()
         .put("permissionName", permission);
       futures.add(doRequest(params, permUrl, HttpMethod.POST, requestBody.encode())
         .compose(response -> {
-          Promise<Boolean> promise = Promise.promise();
           if (response.getCode() == HttpStatus.HTTP_OK.toInt()) {
             LOGGER.info("Added permission {} for {} user", permission, systemUserConfig.getName());
-            promise.complete(true);
+            return Future.succeededFuture();
           } else {
```

Comment:
Also here, there could be cases where mod-permissions returns 400 and that really has to be ignored.

But without proof.. Unlike the one mentioned earlier.

Reply:
Locally I see 422 for this one.

Reply:
You are right, but this is out of scope of this PR. This PR is a code cleanup without changing behavior.
Please create a separate Jira and PR.